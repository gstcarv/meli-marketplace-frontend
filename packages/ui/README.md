# UI

Esse pacote contÃ©m os componentes base para a construÃ§Ãµes dos microfrontends.

## ğŸ“¦ InstalaÃ§Ã£o e ConfiguraÃ§Ã£o

### PrÃ©-requisitos

-   Node.js (versÃ£o 18.18.0+)
-   yarn

### Como rodar o storybook?

1. Instale as dependÃªncias:

    ```bash
    yarn install
    ```

2. Execute o storybook:

    ```bash
    yarn storybook
    ```

## ğŸ§ª Testes

Para rodar os testes, execute:

```bash
yarn test
```
