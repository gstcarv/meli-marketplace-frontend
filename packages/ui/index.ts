import './src/declarations/styled-components.d.ts';

export * from './src/components';

export * from './src/theme';
