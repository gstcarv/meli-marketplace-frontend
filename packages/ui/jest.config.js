module.exports = {
    testEnvironment: 'jsdom',
    transform: {
        '^.+\\.(j|t)sx?$': 'babel-jest'
    },
    moduleNameMapper: {
        '\\.(css)$': 'identity-obj-proxy',
        '\\.(png|jpg|jpeg|gif)$': 'identity-obj-proxy'
    },
    setupFilesAfterEnv: ['@testing-library/jest-dom', '<rootDir>/src/setupTests.ts']
};
