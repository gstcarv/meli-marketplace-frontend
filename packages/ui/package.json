{
    "name": "@meli/ui",
    "version": "0.0.0",
    "private": true,
    "main": "./index.ts",
    "scripts": {
        "lint": "eslint . --max-warnings 0",
        "test": "jest"
    },
    "devDependencies": {
        "@babel/preset-env": "^7.25.8",
        "@babel/preset-react": "^7.25.7",
        "@babel/preset-typescript": "^7.25.7",
        "@meli/design-tokens": "*",
        "@meli/eslint-config": "*",
        "@meli/icons": "*",
        "@meli/typescript-config": "*",
        "@testing-library/jest-dom": "^5.17.0",
        "@testing-library/react": "^12.1.2",
        "@turbo/gen": "^1.12.4",
        "@types/eslint": "^8.56.5",
        "@types/jest": "^27.0.1",
        "@types/node": "^20.11.24",
        "@types/react": "^18.2.61",
        "@types/react-dom": "^18.2.19",
        "babel-jest": "^29.7.0",
        "eslint": "^8.57.0",
        "jest": "^29.7.0",
        "jest-cli": "^29.7.0",
        "jest-environment-jsdom": "^29.7.0",
        "react-test-renderer": "^18.3.1",
        "typescript": "^5.3.3"
    },
    "peerDependencies": {
        "react": "^17.0.2",
        "styled-components": "^6.1.13"
    }
}
