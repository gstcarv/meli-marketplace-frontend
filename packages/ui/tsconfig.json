{
    "extends": "@meli/typescript-config/react-library.json",
    "compilerOptions": {
        "outDir": "dist",
        "module": "ES6",
        "moduleResolution": "Bundler"
    },
    "include": ["src", "./index.ts", "../design-tokens/src/lib"],
    "exclude": ["node_modules", "dist"]
}
