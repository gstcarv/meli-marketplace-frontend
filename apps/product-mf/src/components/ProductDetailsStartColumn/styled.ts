import { breakpoints } from '@meli/design-tokens';
import { Divider } from '@meli/ui';
import styled from 'styled-components';

export const Container = styled.section`
    padding: ${({ theme }) => theme.spacing.get('lg')};
    word-break: break-all;

    @media screen and (max-width: ${breakpoints.get('lg')}) {
        padding: ${({ theme }) => theme.spacing.get('xs')};
    }
`;

export const DescriptionTitle = styled.h2`
    font-size: ${({ theme }) => theme.fontSize.get('h4')};
    margin-bottom: ${({ theme }) => theme.spacing.get('lg')};
    font-weight: ${({ theme }) => theme.fontWeight.get('medium')};

    @media screen and (max-width: ${breakpoints.get('lg')}) {
        font-size: ${({ theme }) => theme.fontSize.get('h6')};
        margin-top: ${({ theme }) => theme.spacing.get('xl')};
        margin-bottom: ${({ theme }) => theme.spacing.get('sm')};
    }
`;

export const DescriptionText = styled.p`
    font-size: ${({ theme }) => theme.fontSize.get('h6')};
    font-weight: ${({ theme }) => theme.fontWeight.get('regular')};
    color: ${({ theme }) => theme.colors.get('gray-600')};

    @media screen and (max-width: ${breakpoints.get('lg')}) {
        font-size: ${({ theme }) => theme.fontSize.get('body')};
    }
`;

export const PictureDivider = styled(Divider)`
    margin: ${({ theme }) => theme.spacing.get('lg')} 0;

    @media screen and (max-width: ${breakpoints.get('lg')}) {
        margin: ${({ theme }) => theme.spacing.get('lg')} 0;
    }
`;
