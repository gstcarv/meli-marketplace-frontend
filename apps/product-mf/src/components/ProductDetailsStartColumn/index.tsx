import { ProductItem } from '@marketplace/bff';
import { ProductImage } from '../ProductImage';
import { Container, DescriptionText, DescriptionTitle, PictureDivider } from './styled';
import { Divider } from '@meli/ui';

type Props = {
    product: ProductItem;
};

const ProductDetailsStartColumn = ({ product }: Props) => {
    return (
        <>
            <Container className='ProductDetails__picture'>
                <ProductImage picture={product?.picture} />

                <PictureDivider />
            </Container>

            <Container className='ProductDetails__info'>
                <section>
                    <DescriptionTitle>Descrição do Produto</DescriptionTitle>

                    <DescriptionText
                        dangerouslySetInnerHTML={{
                            __html: product?.description || ''
                        }}
                    />
                </section>
            </Container>
        </>
    );
};

export default ProductDetailsStartColumn;
