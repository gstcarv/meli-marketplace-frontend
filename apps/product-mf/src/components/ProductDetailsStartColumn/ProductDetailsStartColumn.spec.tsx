import { ProductItem } from '@marketplace/bff';
import { ThemeProvider } from '@meli/ui';
import { render, screen } from '@testing-library/react';
import ProductDetailsStartColumn from '.';

const productStub: ProductItem = {
    id: 'mock-id',
    condition: 'new',
    title: 'Mocked product title',
    price: {
        amount: 4445,
        currency: 'ARS'
    },
    free_shipping: true,
    picture: 'http://mock/picture.png',
    description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec elementum interdum sapien, vitae consectetur diam dictum condimentum. Nunc nec tellus nisl. Nulla ullamcorper, dui quis aliquet ultrices, magna velit condimentum elit, quis blandit mauris nunc sit amet justo. Donec at turpis magna. Suspendisse potenti ',
    sold_quantity: 100
};

describe('<ProductDetailsStartColumn />', () => {
    it('should render', () => {
        const { container } = render(
            <ThemeProvider>
                <ProductDetailsStartColumn product={{ ...productStub, sold_quantity: 30000, condition: 'new' }} />
            </ThemeProvider>
        );

        expect(container).toMatchSnapshot();
    });
});
