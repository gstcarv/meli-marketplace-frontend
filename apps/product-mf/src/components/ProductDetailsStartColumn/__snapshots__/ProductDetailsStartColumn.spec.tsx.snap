// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetailsStartColumn /> should render 1`] = `
.c2 {
  border: 0;
  height: 1px;
  background: #EEEEEE;
  width: 100%;
  margin: 0 0;
}

.c1 {
  width: 100%;
  height: 500px;
  border-radius: 0.25rem;
  display: flex;
  overflow: hidden;
  flex-shrink: 0;
}

.c1:empty {
  background: #EEEEEE;
}

.c1 img {
  width: 100%;
  height: 100%;
  object-fit: contain;
  object-position: center;
}

.c0 {
  padding: 1.5rem;
}

.c4 {
  font-size: 1.375rem;
  margin-bottom: 1.5rem;
  font-weight: 400;
}

.c5 {
  font-size: 1.125rem;
  font-weight: 300;
  color: #727272;
}

.c3 {
  margin: 1.5rem 0;
}

@media screen and (max-width: 992px) {
  .c1 {
    height: 200px;
  }
}

@media screen and (max-width: 992px) {
  .c0 {
    padding: 0.5rem;
  }
}

@media screen and (max-width: 992px) {
  .c4 {
    font-size: 1.125rem;
    margin-top: 2rem;
    margin-bottom: 0.75rem;
  }
}

@media screen and (max-width: 992px) {
  .c5 {
    font-size: 1rem;
  }
}

@media screen and (max-width: 992px) {
  .c3 {
    margin: 1.5rem 0;
  }
}

<div>
  <section
    class="c0 ProductDetails__picture"
  >
    <div
      class="c1"
    >
      <img
        loading="lazy"
        src="http://mock/picture.png"
      />
    </div>
    <hr
      class="c2 c3"
    />
  </section>
  <section
    class="c0 ProductDetails__info"
  >
    <section>
      <h2
        class="c4"
      >
        Descrição do Produto
      </h2>
      <p
        class="c5"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec elementum interdum sapien, vitae consectetur diam dictum condimentum. Nunc nec tellus nisl. Nulla ullamcorper, dui quis aliquet ultrices, magna velit condimentum elit, quis blandit mauris nunc sit amet justo. Donec at turpis magna. Suspendisse potenti 
      </p>
    </section>
  </section>
</div>
`;
